.github/workflows/src/summarize-checks/summarize-checks.js

```diff
@@ -473,6 +473,55 @@ export async function summarizeChecksImpl(
   core.summary.write();
 }
 
+/**
+ * Detect if multiple PRs share the same commit SHA and return a failing protection run.
+ * @param {import('@actions/github-script').AsyncFunctionArguments['github']} github
+ * @param {string} owner
+ * @param {string} repo
+ * @param {number} prNumber
+ * @returns {Promise<CheckRunData|undefined>}
+ */
+export async function detectDuplicatePR(github, owner, repo, prNumber) {
+  const { data: pr } = await github.rest.pulls.get({ owner, repo, pull_number: prNumber });
+  const headRef = pr.head.ref;
+  if (!pr.head || !pr.head.repo) {
+    console.warn(
+      `Cannot determine head branch repo for PR #${prNumber}, skipping duplicate PR detection.`,
+    );
+    return undefined;
+  }
+  const headOwner = pr.head.repo.owner.login;
+  const relatedPRs = await github.paginate(github.rest.pulls.list, {
```

Comment:
I wonder how `github.rest.pulls.list` (passing the head SHA) differs from `github.rest.repos.listPullRequestsAssociatedWithCommit`?

Reply:
We should extract the existing code from `context.js` and create a helper method `listPullRequestsForCommit(head_owner, head_repo, head_sha, base_owner, base_repo)`, which tries the following:

1. repos.listPullRequestsAssociatedWithCommit(head_owner, head_repo, head_sha)
    - Should work on fork and non-fork PRs to public, and non-fork PRs to private
2. pulls.list(base_owner, base_repo, orderby: updated-desc), client-side filter by head_sha
    - Should work on fork PRs to private
    - Max 100 pulls per API call, so maybe only search first page?
3. search(base_owner, base_repo, head_sha)
    - Should work on fork and non-fork PRs to public and private
    - Low rate limit of 30 req/min

Reply:
https://github.com/Azure/azure-rest-api-specs/issues/36853